ote/src/cljs/ote/views/main.cljs

```diff
@@ -29,25 +29,50 @@
     "active")
   )
 
- (defn main-menu [e!]
-   [ui/icon-menu
-    {
+ (defn user-menu [e! user-name]
+   [ui/select-field
+    {:value 0 ;; Force user-name to be selected at all times
+     :label-style {:color "#f2f2f2"}
      :on-click #(e! (fp-controller/->OpenUserMenu))
-     :icon-button-element (reagent/as-element [ui/icon-button [ic/action-view-headline {:color :white}]])
      :anchor-origin {:horizontal "right" :vertical "bottom"}
      :target-origin {:horizontal "right" :vertical "top"}
      }
-    [ui/menu-item {:primary-text (tr [:common-texts :header-front-page])
+    [ui/menu-item {:value 0 ;; First and only element that can be selected in this select field
+                   :primary-text user-name
+                   :selected-text-color (color :grey900)
+                   :on-click #(e! (fp-controller/->ChangePage :front-page))} ]
+    [ui/menu-item {:primary-text (tr [:common-texts :user-menu-service-guide])
+                   :selected-text-color (color :grey900)
                    :on-click #(e! (fp-controller/->ChangePage :front-page))} ]
     [ui/menu-item {:primary-text "Organisaation perustiedot"
+                   :selected-text-color (color :grey900)
                    :on-click #(e! (fp-controller/->ChangePage :transport-operator))} ]
     [ui/menu-item {:primary-text " Näytä debug state"
+                   :selected-text-color (color :grey900)
                    :on-click #(e! (fp-controller/->ToggleDebugState))} ]
     [ui/menu-item {:primary-text "Kirjaudu ulos"
+                   :selected-text-color (color :grey900)
                    :on-click #(e! (fp-controller/->ChangePage :front-page))} ]
     ]
   )
 
+(defn- create-top-nav
```

Comment:
this fn returns the top-nav component and should be named `top-nav` and called as a component `[top-nav e! app]`

Reply:
Fixed